src/index.ts

```diff
@@ -385,18 +396,78 @@ export async function cli(args: string[]) {
     namedExports: undefined,
     webDependencies: undefined,
   };
-  const doesWhitelistExist = !!webDependencies;
-  const arrayOfDeps = webDependencies || Object.keys(pkgManifest.dependencies || {});
+  let doesWhitelistExist = !!webDependencies;
+  let arrayOfDeps = webDependencies || Object.keys(pkgManifest.dependencies || {});
   const hasBrowserlistConfig =
     !!pkgManifest.browserslist ||
     !!process.env.BROWSERSLIST ||
     fs.existsSync(path.join(cwd, '.browserslistrc')) ||
     fs.existsSync(path.join(cwd, 'browserslist'));
+  let dependencyTree: string | undefined;
+
+  // auto detection with --entry flag
+  if (typeof entry === 'string' && entry.length) {
+    const depSpinner = ora(`Scanning ${entry}`).start();
+    const files = entry.split(',').map(file => path.resolve(cwd, file));
+
+    const timeStart = process.hrtime(); // start perf benchmark
+
+    // grab dependencies
+    const fileDeps = autoResolve(files, cwd);
+
+    // print out dependency tree if specified
+    if (depTree) {
+      dependencyTree = JSON.stringify({dependencies: fileDeps});
+    }
+
+    // this will examine package.json dependencies for any matches.
+    const destRoot = dest.replace(/\/$/, '').split('/'); // take last segment of --dest
```

Comment:
We already do resolution logic inside of  the install() function, I want to avoid duplicating that here. Can we instead just generate a "webDependencies"-like array from our analysis here, and then pass that into install()?

Reply:
Oh I wasn’t aware of that. I can play around with it, and try and pass a “dumber” list to `install()` to see what it’s able to figure out.